testing/integration/upgrade_test.go

```diff
@@ -556,28 +574,17 @@ func checkLegacyAgentHealthAndVersion(t *testing.T, ctx context.Context, f *ates
 
 }
 
-// checkUpgradeWatcherRan asserts that the Upgrade Watcher finished running. We use the
-// presence of the update marker file as evidence that the Upgrade Watcher is still running
-// and the absence of that file as evidence that the Upgrade Watcher is no longer running.
-func checkUpgradeWatcherRan(t *testing.T, agentFixture *atesting.Fixture, fromVersion *version.ParsedSemVer) {
+// waitForUpgradeWatcherToComplete asserts that the Upgrade Watcher finished running.
+func waitForUpgradeWatcherToComplete(t *testing.T, f *atesting.Fixture, fromVersion *version.ParsedSemVer, timeout time.Duration) {
 	t.Helper()
 
 	if fromVersion.Less(*version_8_9_0_SNAPSHOT) {
-		t.Logf("Version %q is too old for a quick update marker check, skipping...", fromVersion)
-		return
+		t.Logf("Version %q is too old for a quick update marker check", fromVersion)
+		timeout = defaultWatcherDuration
 	}
 
-	t.Log("Waiting for upgrade watcher to finish running...")
-
-	updateMarkerFile := filepath.Join(agentFixture.WorkDir(), "data", ".update-marker")
-	require.FileExists(t, updateMarkerFile)
```

Comment:
I tried polling for this file to stop existing, but it never exists at the end of the test. I am actually still a bit confused about why the watcher is still running without this (maybe the path here was wrong?), but waiting for exactly the configured watcher timeout fixes all the tests.

My priority now is to get the tests back into a good state.